Trell.Test/Engine/EngineTest.cs

```diff
@@ -90,7 +123,34 @@ public void Dispose() {
 }
 
 public class EngineTest(EngineFixture engineFixture) : IClassFixture<EngineFixture> {
+    record RequestCtx(RequestCtx.RequestRecord Request) {
+        internal record RequestRecord(string Url, string Method, ImmDict<string, string> Headers, string Body);
+        // This subclass exists because JSON can't handle deserializing an ImmDict (it has multiple constructors),
+        // so instead we deserialize into this object and afterward convert its Dictionary into ImmDict.
```

Comment:
Ah, right, that's because System.Text.Json can't deserialize ImmDicts, but Newtonsoft.Json can.